### ISSUE TITLE
Minion blackout periods

### ISSUE THREAD

**Issue by cro** [saltstack/salt#24929]
Add a config option/grain/pillar item that would prevent minions from running states when present.  Some users would like to ensure that no state runs will occur during "blackout" periods.

**Comment by garethgreenaway** [comment]
Should be able to build on existing code for disabling states and ideas from the scheduler to accomplish this.

**Comment by garethgreenaway** [comment]
@cro Did you already start working on this?

**Comment by cro** [comment]
@garethgreenaway No, I haven't dug in yet.  I was considering a minimalist approach--just a yes or no flag, and then letting folks use the scheduler to set or clear it according to their needs.

**Comment by garethgreenaway** [comment]
@cro Taking that approach the code already exists :)  Using the scheduler to fire the state.enable or state.disable module commands.

**Comment by meggiebot** [comment]
Committed for Boron

**Comment by meggiebot** [comment]
Talk to Tom for implementation details

**Comment by basepi** [comment]
Should use an on-disk "queue" of skipped jobs, plus new execution module to look at, pop from, and run said "queue".

**Comment by basepi** [comment]
As we're a little bit short on time before Boron, I plan to implement this without the queue ideas to start, as that's all the customer requested.

Still thinking about what all will be disabled by the noupdate setting. We need to make sure the minion can be updated to remove that setting. May disable everything except the functions related to however that setting will be handled. (Still not sure whether I should do it in the scheduler, grains, or pillar. We shall see when I dig into this.)

**Comment by basepi** [comment]
Naive implementation in #30868. Once that is merged I will close this issue. I have put the additional functionality request into a new issue, #30872.
